napari/_qt/qt_viewer.py

```diff
@@ -228,6 +229,23 @@ def screenshot(self):
         img = self.canvas.native.grabFramebuffer()
         return QImg2array(img)
 
+    def _save_screenshot(self):
```

Comment:
What do you think about allowing users the option to pass in a file name as well (with the default behavior being as it is now)? I could see `_save_screenshot()` being a useful method for people who want to script screen shots (e.g., load in data set, move to a specific view, then take a screen shot) and I think being able to set the filename would be nice for that use case.

Reply:
Maybe it would be better to have the option of passing in a filename to `screenshot()` itself? Seems simpler overall for users to remember. What do you think about using the same structure @sofroniewn uses here with the `file` kwarg in `to_svg()` https://github.com/napari/napari/blob/7a04984472db9a0b74c4efc26d56a6f56b4ce9dd/napari/layers/base/base.py#L657
```python
        if file:
            # Save svg to file
            with open(file, 'w') as f:
                f.write(svg)

        return svg
```

Reply:
Adding a 'filename' kwarg to `Viewer.screenshot()` works for me!

Reply:
That works for me too. We might want to propagate that keyword up to the screenshot method defined here too https://github.com/napari/napari/blob/c01024f151546f7a2ada9727572efb3e65fa4f0f/napari/viewer.py#L64.

Reply:
We're starting to use `path` quite a bit in the codebase, maybe a better choice than `file` or `filename` and something we might want to change in our `to_svg` methods